src/app/app.component.ts

```diff
@@ -71,20 +71,21 @@ export class AppComponent implements OnInit, OnDestroy {
         );
         this.sub = this.notificationService.numberOfNotifications$.subscribe(
             numberOfNotifications => {
-                setTimeout(() => {
-                    this._numberOfNotifications = numberOfNotifications;
-                });
+                this._numberOfNotifications = numberOfNotifications;
             }
         );
 
-        const newNotificationSubscription = this.notificationService.newNotification$.subscribe(
-            () => {
-                this.notificationBlink = 'rgba(255, 255, 255, 0.5)';
-                setTimeout(() => {
+        const newNotificationSubscription = this.notificationService.newNotification$
+            .pipe(
+                tap(() => {
+                    this.notificationBlink = 'rgba(255, 255, 255, 0.5)';
+                }),
+                delay(150),
+                tap(() => {
```

Comment:
Seems the _of_ tap is being reset 150ms after the start, regardless of a new notification coming, which can give issues if multiple notifications come up in the mean time. May I suggest a `switchMap` which _starts_ at subscription and then returns a `timer` observable to reset it but which will be unsubscribed if a new notification comes?

Reply:
Something like:
```
const newNotificationSubscription = this.notificationService.newNotification$
  .pipe(
    switchMap(() => ((this.notificationBlink = 'rgba...'), timer(150))),
  ).subscribe(() => (this.notificationBlink = 'black'));

Reply:
Or even better with `debounceTime`:
```
const newNotificationSubscription = this.notificationService.newNotification$
  .pipe(
    tap(() => (this.notificationBlink = 'rgba...')),
    debounceTime(150),
  ).subscribe(() => (this.notificationBlink = 'black'));
```

Reply:
That's way cleaner. I'll go with `debounceTime`. Thanks a lot!